diff -Nur enlightenment-0.16.999.041+cvs20071025/src/bin/e_config.c enlightenment-0.16.999.041+cvs20071025.new/src/bin/e_config.c
--- enlightenment-0.16.999.041+cvs20071025/src/bin/e_config.c	2007-10-21 11:23:49.000000000 +0200
+++ enlightenment-0.16.999.041+cvs20071025.new/src/bin/e_config.c	2007-10-26 13:39:11.000000000 +0200
@@ -61,6 +61,11 @@
 	snprintf(buf, sizeof(buf), "%s/.e/e/config/profile.cfg",
 		 homedir);
 	ef = eet_open(buf, EET_FILE_MODE_READ);
+    if (!ef)
+      {
+         snprintf(buf, sizeof(buf), "%s/data/config/profile.cfg", e_prefix_data_get());
+         ef = eet_open(buf, EET_FILE_MODE_READ);
+      }
 	if (ef)
 	  {
 	     char *data;
@@ -1796,6 +1801,11 @@
    snprintf(buf, sizeof(buf), "%s/.e/e/config/%s/%s.cfg",
 	    homedir, _e_config_profile, domain);
    ef = eet_open(buf, EET_FILE_MODE_READ);
+   if (!ef)
+     {
+    snprintf(buf, sizeof(buf), "%s/data/config/%s/%s.cfg", e_prefix_data_get(), _e_config_profile, domain);
+    ef = eet_open(buf, EET_FILE_MODE_READ);
+     }
    if (ef)
      {
 	data = eet_data_read(ef, edd, "config");
