Index: src/modules/conf/e_conf.c
===================================================================
--- src/modules/conf/e_conf.c	(revision 40507)
+++ src/modules/conf/e_conf.c	(revision 40508)
@@ -514,8 +514,8 @@
 
    if (!(eco = data)) return 1;
    if (!eco->cat_list) return 1;
-   sel = e_widget_ilist_selected_get(eco->cat_list);
+//   sel = e_widget_ilist_selected_get(eco->cat_list);
    _e_configure_fill_cat_list(eco);
-   e_widget_ilist_selected_set(eco->cat_list, sel);
+//   e_widget_ilist_selected_set(eco->cat_list, sel);
    return 1;
 }
