diff --git a/src/bin/Makefile.am b/src/bin/Makefile.am
index faef192..8d4b2ba 100644
--- a/src/bin/Makefile.am
+++ b/src/bin/Makefile.am
@@ -340,8 +340,7 @@ enlightenment_thumb_LDFLAGS = @e_libs@
 enlightenment_fm_SOURCES = \
 e_fm_main.c \
 e_user.c \
-e_sha1.c \
-e_prefix.c
+e_sha1.c
 
 enlightenment_fm_LDFLAGS = @e_libs@ @dlopen_libs@
 enlightenment_fm_CFLAGS = $(INCLUDES)
diff --git a/src/bin/Makefile.in b/src/bin/Makefile.in
index 6ac3d48..9775187 100644
--- a/src/bin/Makefile.in
+++ b/src/bin/Makefile.in
@@ -120,8 +120,7 @@ enlightenment_LINK = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) \
 	$(enlightenment_LDFLAGS) $(LDFLAGS) -o $@
 am_enlightenment_fm_OBJECTS = enlightenment_fm-e_fm_main.$(OBJEXT) \
 	enlightenment_fm-e_user.$(OBJEXT) \
-	enlightenment_fm-e_sha1.$(OBJEXT) \
-	enlightenment_fm-e_prefix.$(OBJEXT)
+	enlightenment_fm-e_sha1.$(OBJEXT)
 enlightenment_fm_OBJECTS = $(am_enlightenment_fm_OBJECTS)
 enlightenment_fm_LDADD = $(LDADD)
 enlightenment_fm_LINK = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) \
@@ -709,8 +708,7 @@ enlightenment_thumb_LDFLAGS = @e_libs@
 enlightenment_fm_SOURCES = \
 e_fm_main.c \
 e_user.c \
-e_sha1.c \
-e_prefix.c
+e_sha1.c
 
 enlightenment_fm_LDFLAGS = @e_libs@ @dlopen_libs@
 enlightenment_fm_CFLAGS = $(INCLUDES)
@@ -976,7 +974,6 @@ distclean-compile:
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/e_xinerama.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/e_zone.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/enlightenment_fm-e_fm_main.Po@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/enlightenment_fm-e_prefix.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/enlightenment_fm-e_sha1.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/enlightenment_fm-e_user.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/enlightenment_fm_op-e_fm_op.Po@am__quote@
@@ -1046,20 +1043,6 @@ enlightenment_fm-e_sha1.obj: e_sha1.c
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
 @am__fastdepCC_FALSE@	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(enlightenment_fm_CFLAGS) $(CFLAGS) -c -o enlightenment_fm-e_sha1.obj `if test -f 'e_sha1.c'; then $(CYGPATH_W) 'e_sha1.c'; else $(CYGPATH_W) '$(srcdir)/e_sha1.c'; fi`
 
-enlightenment_fm-e_prefix.o: e_prefix.c
-@am__fastdepCC_TRUE@	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(enlightenment_fm_CFLAGS) $(CFLAGS) -MT enlightenment_fm-e_prefix.o -MD -MP -MF $(DEPDIR)/enlightenment_fm-e_prefix.Tpo -c -o enlightenment_fm-e_prefix.o `test -f 'e_prefix.c' || echo '$(srcdir)/'`e_prefix.c
-@am__fastdepCC_TRUE@	mv -f $(DEPDIR)/enlightenment_fm-e_prefix.Tpo $(DEPDIR)/enlightenment_fm-e_prefix.Po
-@AMDEP_TRUE@@am__fastdepCC_FALSE@	source='e_prefix.c' object='enlightenment_fm-e_prefix.o' libtool=no @AMDEPBACKSLASH@
-@AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-@am__fastdepCC_FALSE@	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(enlightenment_fm_CFLAGS) $(CFLAGS) -c -o enlightenment_fm-e_prefix.o `test -f 'e_prefix.c' || echo '$(srcdir)/'`e_prefix.c
-
-enlightenment_fm-e_prefix.obj: e_prefix.c
-@am__fastdepCC_TRUE@	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(enlightenment_fm_CFLAGS) $(CFLAGS) -MT enlightenment_fm-e_prefix.obj -MD -MP -MF $(DEPDIR)/enlightenment_fm-e_prefix.Tpo -c -o enlightenment_fm-e_prefix.obj `if test -f 'e_prefix.c'; then $(CYGPATH_W) 'e_prefix.c'; else $(CYGPATH_W) '$(srcdir)/e_prefix.c'; fi`
-@am__fastdepCC_TRUE@	mv -f $(DEPDIR)/enlightenment_fm-e_prefix.Tpo $(DEPDIR)/enlightenment_fm-e_prefix.Po
-@AMDEP_TRUE@@am__fastdepCC_FALSE@	source='e_prefix.c' object='enlightenment_fm-e_prefix.obj' libtool=no @AMDEPBACKSLASH@
-@AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-@am__fastdepCC_FALSE@	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(enlightenment_fm_CFLAGS) $(CFLAGS) -c -o enlightenment_fm-e_prefix.obj `if test -f 'e_prefix.c'; then $(CYGPATH_W) 'e_prefix.c'; else $(CYGPATH_W) '$(srcdir)/e_prefix.c'; fi`
-
 enlightenment_fm_op-e_fm_op.o: e_fm_op.c
 @am__fastdepCC_TRUE@	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(enlightenment_fm_op_CFLAGS) $(CFLAGS) -MT enlightenment_fm_op-e_fm_op.o -MD -MP -MF $(DEPDIR)/enlightenment_fm_op-e_fm_op.Tpo -c -o enlightenment_fm_op-e_fm_op.o `test -f 'e_fm_op.c' || echo '$(srcdir)/'`e_fm_op.c
 @am__fastdepCC_TRUE@	mv -f $(DEPDIR)/enlightenment_fm_op-e_fm_op.Tpo $(DEPDIR)/enlightenment_fm_op-e_fm_op.Po
