diff --git a/src/modules/everything/evry_plug_apps.c b/src/modules/everything/evry_plug_apps.c
index 5f7f59c..d979382 100644
--- a/src/modules/everything/evry_plug_apps.c
+++ b/src/modules/everything/evry_plug_apps.c
@@ -1327,7 +1327,7 @@ _conf_new(void)
 
     /* setup defaults */
     IFMODCFG(0x009d);
-    _conf->cmd_terminal = eina_stringshare_add("/usr/bin/xterm");
+    _conf->cmd_terminal = eina_stringshare_add("/usr/bin/x-terminal-emulator");
     _conf->cmd_sudo = eina_stringshare_add("/usr/bin/gksudo --preserve-env");
     IFMODCFGEND;
 
