diff --git a/src/modules/everything-apps/e_mod_main.c b/src/modules/everything-apps/e_mod_main.c
index 0085502..c6a70e7 100644
--- a/src/modules/everything-apps/e_mod_main.c
+++ b/src/modules/everything-apps/e_mod_main.c
@@ -1190,7 +1190,7 @@ _conf_new(void)
 
    /* setup defaults */
    IFMODCFG(0x009d);
-   _conf->cmd_terminal = eina_stringshare_add("/usr/bin/xterm");
+   _conf->cmd_terminal = eina_stringshare_add("/usr/bin/x-terminal-emulator");
    _conf->cmd_sudo = eina_stringshare_add("/usr/bin/gksudo --preserve-env");
    IFMODCFGEND;
 
