Description: Fix upstream bug resetting LDFLAGS.
Author: Andreas Metzler <ametzler@debian.org>
Origin: vendor
Last-Update: 2016-12-21

--- e17-0.21.5.orig/configure.ac
+++ e17-0.21.5/configure.ac
@@ -1041,7 +1041,7 @@ case "$host_os" in
              AC_MSG_RESULT(yes)],
              AC_MSG_RESULT(no))
         CFLAGS="$OLDCFLAGS"
-        LDFLAGS="$OLDLFDLAGS"
+        LDFLAGS="$OLDLDFLAGS"
       fi
       ;;
 esac
